### DISCUSSION TITLE
Is there a way to add arbitrary player ~crafting/professions/trade-skills?

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#857] [Q&A] [answered]
Is there a way to add arbitrary player ~crafting/professions/trade-skills?

Such as Level 1-300 Cooking or Mining or Blacksmithing; and ~'trade skill experience that levels up said skill? I know there is an NPC type 'crafting' thing going on although it is not as deep of progression as crafting/professions/trade-skills levels as found in some grand scale classic RPGs. If player ~crafting/professions/trade-skills can be implemented can also skill required 'nodes' work as well like requires X ~Mining to collect or something in that vein.

Extra details / examples of this are:
Old School Rune Scape: https://oldschool.runescape.wiki/w/Skills
Vanilla World of Warcraft: https://vanilla-twinhead.twinstar.cz/?spells=11.164 + https://vanilla-twinhead.twinstar.cz/?object=1735 + https://classic.wowhead.com/guides/professions-overview-wow-classic
Classic EverQuest: https://wiki.project1999.com/Skill_Blacksmithing + https://wiki.project1999.com/Tradeskills
Lots of fun and like 1/3 or 1/2+ of these example game's main progression can be found in advancing theses ~crafting/professions/trade-skills. If this can made with its' way into the Flare Engine than we can have ~double the player progression and better integration of world content to player progression. Even a basic template with tab with a list / some kind of basic menu with a list and progress / trade skill level tracking could work wonders. WoW has a 'recipe' list and current trade skill level menu system. Also it works with having a trade skill world object (like an anvil) + inventory item (like a blacksmithing hammer) in order to craft an item at a trade skill 'work site' (like an anvil). Also the ability to learn more recipes via trainers and items. Also Rune Scape has a basic level tracking menu for skills and the the work table / bench / anvil etc. opens up a list of possible items to craft; a bit clunky but it works. Most grand scale RPGs use the WoW / Everquest recipe list and trade skill level tracking menu with trade skill world objects (like an anvil) + inventory items (like a blacksmithing hammer) and the ability to discover more recipes in the world via trainers and consumable recipe items (named like blacksmithing plans, cooking recipes etc. https://classic.wowhead.com/items/recipes ).

**Comment by Danimal696** [comment]
Someone had some success at this, look at Flare forum, someone is doing a Stardew Valley clone that implemented raising skills.

**Reply by WithinAmnesia** [reply]
Didn't that just swap out main attributes with pseudo-skills though? Like it gutted the main game play features for some farm skill functionality? I am hoping for more skills that can have their own menu and are separate from the main combat attributes. Although the rocks and trees and other things taking damage was pretty cool. Although the arbitrary skill requirement is still left open.

**Reply by WithinAmnesia** [reply]
https://opengameart.org/forumtopic/open-valley-ideaproject?page=2 I am not sure this counts but there could be possible tricks in here to learn from; maybe with stuff like 'resource nodes'. Although I suspect that a new system is needed given the core differences of a combat driven main game with added ~trade skills verses a combat gutted system with farming skills as the main focus.

I think there is a way to add more menus and arbitrary skills and levels to the Flare engine but I am not sure how easy / difficult it would be and what options are there for implementation. Right now Flare is without a deep level based arbitrary crafting / trade skill / profession system that could add much more (upwards of ~1/3 or even ~1/2) additional character progression.

I am wondering how to add this kind of successful classic / traditional deep progression system of arbitrary crafting levels / trade skills / professions to this item and character level progression driven R.P.G. in a realistic / reasonable way?

**Comment by WithinAmnesia** [comment]
I remember with World of Warcraft each player character was restricted to 2 main professions / trade skills. This effectively made a game where usually a player would be expected to have to trade with other players to gain the available benefits of each trade skill / professions.

Also Path of Exile (and Diablo 3 and to a lesser extent Baldur's Gate) works where you talk to an Non-Player Character to negotiate a trade of usually materials and or money and or time to have the said N.P.C. craft the player what they requested be made for them via plying the ~trade/skill of the N.P.C. in question.

Also Minecraft has a pretty deep crafting system, but it uses a grid system and is kind of its own thing / m ore of a block game thing / has its limitations. I know Diablo II has a similar crafting system but it feels... kind of limited in many ways (no skill requirements, things magically instantly appear with limit or no 'work bench' minus a few exceptions like in Minecraft smelting requires a forge ans time).

So. I am hoping to figure out how to get those 'juicy' trade skill progression into Flare but without being a nightmare.
I remember in my ~frozen Baldur's Gate Arms and Armour Emporium there is a ~crafting / (re)forging system where the player talks and pays for the services to transform ~materials and wealthy / money into finished ~Arms and Armour after a period of time has elapsed.

So it could be possible to combine the Flare crafting system with my Arms and Armour approach to crafting where as just like in player trades in World of Warcraft or having Non-Player Characters ~craft things for the player like in Path of Exile (and Diablo 3 and bits of Baldur's Gate). This would still have a similar effect where the player can find trade skill related objects in the world and would have to travel to ~civilization to process and craft and or transform these ~base materials into better more useful objects and or items to hopefully benefit the player. Although the biggest thing now would be seeing if the Flare crafting system can work with putting in like ~10,000 currency + other ingredient materials to craft things.

Also having a Non-Player Character be used to access crafting simulates that trades person type of world building and the world can be populated with different kinds of trades people with different trades and hopefully different recipe combinations to offer and incentivizing game world exploration and trade amongst different locations to reap the hopefully divers possibilities of in game economy offered via trades people and different trades people crafting recipe combinations. Also having currency be required to craft items every time makes a nature currency sink to give a reason for the player to spend lots of money at the end game and even possibly have a system like Diablo II gambling where if the outcome is not guaranteed and has a chance to make a desired ~item than that could work great additionally on top of the hopefully diverse and robust crafting ~ecosystem potentially hosted within the Flare Engine.

I should make a different thread with more specif questions following down this rabbit hole of trying to implement deep player progression crafting/professions/trade-skills with the current crafting system within the Flare engine; hmmm.

**Comment by WithinAmnesia** [comment, marked as answer]
Successful Crafting Approach: (It works now!) Via: https://github.com/flareteam/flare-engine/issues/1796#issuecomment-979806515 + https://github.com/flareteam/flare-game/discussions/859

# Craft Dagger Test
[button]
button_pos=248,280
text=Forge Copper Off-Hand Dagger.
requires_currency=100
requires_item=3097:3
script=scripts/AGABAL/Crafting/Blacksmithing/Crafting_Test_Copper_Off-Hand_Dagger.txt

[event]
remove_currency=100
remove_item=3097:3
reward_item=3071
